### PR TITLE
Tests, Travis updates, Cleanup

### LINKED ISSUE #27: Sprint 2 Tests, implement #33's TODO
Tests for sprint 2 components.
Depends on at least #21, #22, #23, and #24 

Implement #33's TODO: replace filler text with actual descriptions of personality outputs

Add Kenny's bug fix from #34

### PR DESCRIPTION
**Pull request by elrodrigues**
Pull request for general tasks: cleanup, refactoring, Kenny's bug fix, adding tests, and output for Jenny's UI.
Closes #27 

Changes:
- @khzheng24998's bug fix from #34 (this will be reviewed in this pull request)
- Implemented #33's TODO
- Swapped `results` and `profile_id`
- Beautified algorithm output
- Added tests for DB and results.html
- Updated `conftest.py`

**TBD**
- Descriptions for personality factors
- ~Update Travis with script to check and initialize DB~ No need
- ~(Optional) add Travis deployment~ (like Heroku? I don't think we need to)

https://user-images.githubusercontent.com/44657527/109253371-424c1380-77b5-11eb-9328-8797830a3e9c.mp4

### CODE REVIEW

**Comment by elrodrigues** [comment]
Working on tests now. Travis may break beyond this point

**Comment by elrodrigues** [comment]
`conftest` creates a temporary test database everytime pytest is run, so we don't have to explicitly create a database with a script

**Comment by elrodrigues** [comment]
Travis should be stable now. However, I foresee `test_results.py` breaking in the future when we switch from sandbox to real API

**Comment by elrodrigues** [comment]
Current coverage for `__init__.py` is 58% from last Travis build. I plan to split test_factory into smaller components that test a page.

File: flaskr/algorithm.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def generate_profile(portfolio, finnhub_client):
         
         E = 19.23*(MB - 1.7364) + 0.0006*(Size + 4124.38)
         I = 18.518*(MB - 1.80656) + 0.0005*(12679.64 - Size)
-        S = 6.757*(2.37625 - MB) + 0.0295*(Size - 4895.96)
+        S = 13.514*(2.37625 - MB)
```
Comment (daniel77chen) [inline flaskr/algorithm.py:55]: just to clarify, this change was made bc Sentimentality is not associated with MB, right?

Comment (elrodrigues) [reply flaskr/algorithm.py:55]: The calculation gave equal weightage to Market-to-Book Ratio and Market Size, but this was producing skewed results due to issues with Market Size data. I figured MB is more consistent, so I gave MB full weightage

Comment (daniel77chen) [reply flaskr/algorithm.py:55]: also, can I ask why we need a cap for the Size at 14e3? (in line 50). thank you ^_^

Comment (elrodrigues) [reply flaskr/algorithm.py:55]: > 
> 
> also, can I ask why we need a cap for the Size at 14e3? (in line 50). thank you ^_^

Just for our convenience. Imagine a company like Amazon with huge market cap that would heavily skew the results. From my (unscientific) math, 14e3 seemed to be the limit. I'm open to changing this.

Comment (daniel77chen) [reply flaskr/algorithm.py:55]: > > also, can I ask why we need a cap for the Size at 14e3? (in line 50). thank you ^_^
> 
> Just for our convenience. Imagine a company like Amazon with huge market cap that would heavily skew the results. From my (unscientific) math, 14e3 seemed to be the limit. I'm open to changing this.

okay got it, I think we can just go with 14e3 for now then. thanks for explaining 👍

**Comment by elrodrigues** [comment]
`__init__.py` is now 74% covered

**Comment by elrodrigues** [comment]
97% covered

**Comment by jenny-l-li** [comment]
lgtm

**Pull request merged**